### PR TITLE
Add sudo extension for extendGraphqlSchema

### PR DESCRIPTION
**Pull request by dcousens**
This pull request adds a `sudo: boolean` GraphQL extension that allows users to differentiate whether they are extending the GraphQL schema for a normal, or `.sudo()` context.

### CODE REVIEW

File: packages/auth/src/schema.ts [hunk 1]
```diff
@@ -58,13 +58,14 @@ export const getSchemaExtension = ({
...
   sessionData: string;
-}): ExtendGraphqlSchema =>
+}) =>
   graphql.extend(base => {
     const uniqueWhereInputType = assertInputObjectType(
       base.schema.getType(`${listKey}WhereUniqueInput`)
     );
     const identityFieldOnUniqueWhere = uniqueWhereInputType.getFields()[identityField];
     if (
+      base.schema.extensions.sudo &&
```
Comment (dcousens) [inline packages/auth/src/schema.ts:68]: @borisno2 I don't think this needs a changeset, no behaviour has changed

File: packages/core/src/lib/createGraphQLSchema.ts [hunk 2]
```diff
@@ -1,41 +1,134 @@
...
+    mutation: mutation.graphQLType,
+    // not about behaviour, only ordering
+    types: [...collectTypes(lists, updateManyByList), mutation.graphQLType],
+    extensions: {
+      sudo,
     },
   });
+}
+
+function collectTypes(
```
Comment (dcousens) [inline packages/core/src/lib/createGraphQLSchema.ts:55]: Unchanged from `packages/core/src/lib/core/graphql-schema.ts`

File: packages/auth/src/index.ts [hunk 3]
```diff
@@ -302,9 +307,9 @@ export function createAuth<ListTypeInfo extends BaseListTypeInfo>({
         ...keystoneConfig.lists,
         [listKey]: { ...listConfig, fields: { ...listConfig.fields, ...fields } },
       },
-      extendGraphqlSchema: existingExtendGraphQLSchema
-        ? schema => existingExtendGraphQLSchema(extendGraphqlSchema(schema))
-        : extendGraphqlSchema,
+      extendGraphqlSchema: schema => {
+        return extendGraphqlSchema(authExtendGraphqlSchema(schema));
+      },
```
Comment (dcousens) [inline packages/auth/src/index.ts:312]: Not part of this pull request, but is equivalent to the same pattern we used elsewhere

**Review by borisno2 — APPROVED** [review]
